### PR TITLE
Add warning about TagBot permissions

### LINKED ISSUE #1277: /stable is always linking to the previous version
I'm building the documentation using github actions, and after TagBot creates a new tag + release, the stable version of the documentation is still the previous one - for example:

https://github.com/EcoJulia/GBIF.jl/runs/550632506?check_suite_focus=true#step:5:32

Is there something I am missing? I have no added options besides `push_preview=true` to the `make.jl`
 file.

### PR DESCRIPTION
**Pull request by mortenpi**
Fix #1277

### CODE REVIEW

File: docs/src/man/hosting.md [hunk 1]
```diff
@@ -214,6 +214,34 @@ information on how to setup a GitHub workflow see the manual for
 The commands in the lines in the `run:` section do the same as for Travis,
 see the previous section.
 
+!!! warning "TagBot & tagged versions"
+
+    In order to deploy documentation for **tagged versions**, the GitHub Actions workflow
+    needs to be triggered by the tag. However, by default, when the [Julia TagBot](https://github.com/marketplace/actions/julia-tagbot)
+    uses just the `GITHUB_TOKEN` for authentication, it does not have the permission to trigger
+    any further workflows jobs, and so the the documentation CI job never runs for the tag.
```
Comment (fredrikekre) [inline docs/src/man/hosting.md:222]:
```suggestion
    any further workflows jobs, and so the documentation CI job never runs for the tag.
```

File: docs/src/man/hosting.md [hunk 2]
```diff
@@ -214,6 +214,34 @@ information on how to setup a GitHub workflow see the manual for
...
+
+    To work around that, TagBot should be [configured to use `DOCUMENTER_KEY`](https://github.com/marketplace/actions/julia-tagbot#ssh-deploy-keys)
+    for authentication, by adding `ssh: ${{ secrets.DOCUMENTER_KEY }}` to the `with` section.
+    A complete TagBot workflow file could look as follows:
+
+    ```yml
+    name: TagBot
+    on:
+      schedule:
+        - cron: 0 * * * *
```
Comment (fredrikekre) [inline docs/src/man/hosting.md:232]:
```suggestion
        - cron: 0 0 * * *
```

**Comment by DilumAluthge** [comment]
Bump @mortenpi @fredrikekre 

It would be great to get this added into the documentation.

**Comment by mortenpi** [comment]
Nothing specifically holding it back actually, but I have been using this PR to see how the recently re-enabled linkchecking tests fare.

**Pull request merged**